Add features for the front end app in the full-stack capstone project - Tesla Car Test Drive Booking

ivangonzalez224 opened the pull request:
## Final PR - Front end app for Full Stack Capstone project - Tesla Car Test Drive Booking

**Description:**

This pull request implements the following features for the Tesla car test drive booking front end app:

- Installed react-redux, react-router, react-icons, axios and the necessary dependencies.
- Added the store and the reducers.
- Added dynamic routing for the whole project.
- Added the components for each page.
- Set up the structure of navigation panel, the sign in, sign out, the main, details, reserve form, my reservations and create and delete car pages.
- Added the functionality to sign in and sign out in a proper way by fetching the authentication APIs.
- Fetched the cars, reservations and cities data from the API.
- Displayed the list of cars on the main page.
- Added logic to filter reservations by user ID and is_remove cars column.
- Rendered the filtered data in my reservations page.
- Added styles to the navigation panel, the main, details, reserve form, my reservations and create and delete car pages.

**Technical Setup:**

Gitflow workflow implemented.
React & Redux for frontend UI.

ivangonzalez224 committed:
Install Redux and Router DOM

ivangonzalez224 committed:
Install linters

ivangonzalez224 committed:
Update the readme file and include the license

hafiz1379 committed:
Merge pull request #24 from ivangonzalez224/feature/initial-set-up

Initial set up

hafiz1379 committed:
Refactor CSS and update package dependencies

hafiz1379 committed:
Remove unnecessary lines in index.html

hafiz1379 committed:
Add axios configuration

hafiz1379 committed:
Refactor routes and components in App.js

hafiz1379 committed:
Refactor index.js to include Redux, React Router,
and AuthProvider

hafiz1379 committed:
Add logo.png file

hafiz1379 committed:
Add carousel CSS styles

hafiz1379 committed:
Add CSS styles for index page

hafiz1379 committed:
Add Login component

hafiz1379 committed:
Add Navbar component

hafiz1379 committed:
Fix bug in login functionality

hafiz1379 committed:
Add RequireAuth component for authentication

hafiz1379 committed:
Add Sign In Form component

hafiz1379 committed:
Add SignUpForm component

hafiz1379 committed:
Add AddCarsForm component

hafiz1379 committed:
Add CarDetails component

hafiz1379 committed:
Add CarItem component

hafiz1379 committed:
Add DeleteCar component

hafiz1379 committed:
Add MainPage component for displaying cars

hafiz1379 committed:
Add Reservations component

hafiz1379 committed:
Add AuthProvider context

hafiz1379 committed:
Add seAuth hook for authentication

hafiz1379 committed:
Add Redux store configuration

hafiz1379 committed:
Add carsSlice reducer and async thunks

hafiz1379 committed:
Fix bug in login functionality

hafiz1379 committed:
Fix bug in login functionality

hafiz1379 committed:
Update eslint and eslint-config-airbnb

hafiz1379 committed:
Fix bug in login functionality

hafiz1379 committed:
Remove commented code and update function syntax

hafiz1379 committed:
Refactor Login component

hafiz1379 committed:
Refactor Register component styling

hafiz1379 committed:
Refactor RequireAuth component

hafiz1379 committed:
Refactor SignInForm component

hafiz1379 committed:
Refactor SignUpForm component and update API
endpoint

hafiz1379 committed:
Refactor AddCarsForm component

hafiz1379 committed:
Refactor CarItem component to use function syntax

hafiz1379 committed:
Refactor MainPage component

hafiz1379 committed:
Fix linting issue in AuthProvider.js

hafiz1379 committed:
Update logout functionality and license text

hafiz1379 committed:
Merge pull request #25 from ivangonzalez224/feature/login

Feature/login

kambereBr committed:
Export function getAllCars

kambereBr committed:
Add carsReducer

kambereBr committed:
Call and dispatch all cars function on main page loading

kambereBr committed:
Load and didplays the list of all cars

kambereBr committed:
Fix linters

kambereBr committed:
Merge pull request #26 from ivangonzalez224/feature/main_page

🎖️ Create main page 🎖️

ivangonzalez224 committed:
Update route paths and create the Session component

ivangonzalez224 committed:
Add the Session component structure

ivangonzalez224 committed:
Add styles to the session component for mobile version

ivangonzalez224 committed:
Add styles to the Session component for desktop version

ivangonzalez224 committed:
Update the add car form structure

ivangonzalez224 committed:
Add styles to the add new car form

StevenWafeek committed:
Merge pull request #27 from ivangonzalez224/feature/add-car

Add car

StevenWafeek committed:
Add car details page

StevenWafeek committed:
Update app file

StevenWafeek committed:
Add the css page

StevenWafeek committed:
Update car item file

StevenWafeek committed:
fix linters

StevenWafeek committed:
Fix linters

StevenWafeek committed:
Update css

StevenWafeek committed:
Add reserved button

StevenWafeek committed:
Update css file

StevenWafeek committed:
Fix linters

ivangonzalez224 committed:
Merge pull request #28 from ivangonzalez224/feature/details_page

Feature/details page

ivangonzalez224 committed:
Update the DeleteCar component to show a table with the available cars list

ivangonzalez224 committed:
Update the carRemoved reducer to make a car no longer available

ivangonzalez224 committed:
Create the handleDeleteCar function to update the state and the available cars table

ivangonzalez224 committed:
Include the handleCarRemoved function in the Main page to update the car list

ivangonzalez224 committed:
Create the handleUpdateCar function to update the is-removed column in the cars table

ivangonzalez224 committed:
Create the updateCars extra reducer to perform the delete method to update the is_removed cars column

ivangonzalez224 committed:
Add styles to the available cars table

ivangonzalez224 committed:
Merge pull request #29 from ivangonzalez224/feature/delete-car

Delete car

kambereBr committed:
Add react-multi-carousel package

kambereBr committed:
Implement carousel component to MainPage.jsx

kambereBr committed:
Refactor CarItem component to remove inline style

kambereBr committed:
Add carousel styles to CSS file

kambereBr committed:
Refactor index.css file to fit best practices

kambereBr committed:
Merge branch 'dev' into feature/carousel

kambereBr committed:
Fix linters error

kambereBr committed:
fix linters in mainPage component

kambereBr committed:
Merge pull request #30 from ivangonzalez224/feature/carousel

🎖️ Enhance carousel style 🎖️

Alibaba2023 committed:
available the links to delete and create cars for admin users

Alibaba2023 committed:
Merge pull request #31 from ivangonzalez224/feature_authorization

Available the links to delete and create cars for admin users

ivangonzalez224 committed:
Include the reserve form option in the nav panel

ivangonzalez224 committed:
Create the ReserveNav component and the select car

ivangonzalez224 committed:
Add the inputs in the ReserveNav component

StevenWafeek committed:
Update css file

StevenWafeek committed:
Update reserveNav

StevenWafeek committed:
Update reserveNav

StevenWafeek committed:
Add cityservice file

StevenWafeek committed:
Update reserveNav file

ivangonzalez224 committed:
Update the handleSubmit function to create a new reservation

ivangonzalez224 committed:
Create the cities slice to get a list of cities from the cities API

ivangonzalez224 committed:
Fetch the cities list from the main page

kambereBr committed:
Fix navigation bug in CarItem component

kambereBr committed:
Add console logs and update reserved button link
in CarDetails.js

kambereBr committed:
Add city dropdown and reservation form to
ReserveDetails component

Alibaba2023 committed:
fixed the reservation of the car

Alibaba2023 committed:
Go Back button added in the reservationform

kambereBr committed:
Handle Go Back buttons properly during reservation process

kambereBr committed:
Fix linters

StevenWafeek committed:
Add css reservation file

StevenWafeek committed:
Update reserve pages

StevenWafeek committed:
Update css

StevenWafeek committed:
Fix linters

ivangonzalez224 committed:
Adjust styles to meet the required design

ivangonzalez224 committed:
Merge pull request #32 from ivangonzalez224/feature/reserve-form

Reserveation form

StevenWafeek committed:
Update carousel style

StevenWafeek committed:
Update main page style

StevenWafeek committed:
Update details page styles

StevenWafeek committed:
Add animation to details p

StevenWafeek committed:
Fix linters

StevenWafeek committed:
Update login page styles

kambereBr committed:
Add reservationsSlice.js to handle reservations in Redux

kambereBr committed:
Add reservationsReducer to store

kambereBr committed:
Add reservation list to Reservations component

kambereBr committed:
Fix linters

kambereBr committed:
Update reservation title

StevenWafeek committed:
Update styles and cardetails register and logn files

StevenWafeek committed:
Merge pull request #33 from ivangonzalez224/feature/responsive

Make the app responsive

kambereBr committed:
Merge branch 'dev' into feature/list_reservation

kambereBr committed:
Update Delete car and My reservations styling

kambereBr committed:
Update author information in README.md

kambereBr committed:
Update navigation in Login component

kambereBr committed:
Update carousel styles and add arrow buttons styles

kambereBr committed:
Refactor and Remove SignInForm and SignUpForm components

kambereBr committed:
Fix formatting in carouselStyle.css

kambereBr committed:
Merge pull request #34 from ivangonzalez224/feature/list_reservation

🎖️ My reservation feature 🎖️

Meltrust commented on src/App.js:
```diff
@@ -1,24 +1,33 @@
+import ReserveNav from './components/reserve/ReserveNav';
+import ReserveDetails from './components/reserve/ReserveDetails';
 
 function App() {
```
- [ ] This is the test that comes with React:

![image](https://github.com/ivangonzalez224/Tesla-car-booking-front-end/assets/57421823/ed060300-33ee-463a-ad18-3ca1a19f9aa8)

Kindly either implement it or remove it (the file) if not using it. 👍

Meltrust replied:
- [ ]  Please, kindly refactor this main "App" functional component to ES6 arrow function, so your code complies with the latest standards.

Refactoring is easy:

```javascript
const App = () => {
  // ...
}
```

## Explanation

The `function` keyword has been soft deprecated in almost all the software industry in favor of ES6 arrow functions. This means, almost nobody uses it anymore, so much that it's considered a non-best practice.

That way, your code will meet the latest standards. 💪

Meltrust commented on src/App.js:
```diff
@@ -1,24 +1,33 @@
-import logo from './logo.svg';
-import './App.css';
+import React from 'react';
```
- [ ] When adding a new car, the image is inconsistent with the others:

![image](https://github.com/ivangonzalez224/Tesla-car-booking-front-end/assets/57421823/6b1083ed-1309-4bc1-b47a-a4283ed10762)

Kindly, add some boundaries to your CSS styling, so any user added image is the same size as the others. 👍

Meltrust reviewed (changes requested):
Hi,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Nice validations ✔️ 



#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

![You can do it](https://media.giphy.com/media/FemTGpEZeK27M5fKL4/giphy.gif)

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

StevenWafeek committed:
Add the img container to css

StevenWafeek committed:
Update details page css

StevenWafeek committed:
Fix linters

StevenWafeek committed:
Remove the test file

StevenWafeek committed:
Update app file with arrow function

StevenWafeek committed:
Update the car list styles

StevenWafeek committed:
Merge pull request #36 from ivangonzalez224/feature/fix-responsive

Feature/fix responsive

ivangonzalez224 commented:
Hi @Meltrust thank you for the review
Changes done: refactored the app component to meet the latest standards, removed the test file and fixed the styles in Main page.

Meltrust reviewed (approved):
# **Wow, you did it 🎉**
![Brilliant](https://media.giphy.com/media/d31w24psGYeekCZy/giphy.gif)
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
![applause](https://media.giphy.com/media/srg19CG0cKMuI/giphy.gif)

### To highlight


- Superb job ✔️  



Cheers and Happy coding!👏👏👏


------

Alibaba2023 committed:
MainPage function changed

Alibaba2023 committed:
Changed the DeleteCar function to arrow function

Alibaba2023 committed:
Changed the CarItem function to arrow function

Alibaba2023 committed:
Changed the CarDetails function to arrow function

Alibaba2023 committed:
Changed the Reservations function to arrow function

Alibaba2023 committed:
Changed the Navbar function to arrow function

Alibaba2023 committed:
Kanban board section added

ivangonzalez224 committed:
Include the final team size in the readme file

Alibaba2023 committed:
Merge pull request #37 from ivangonzalez224/feature/refactor_code

Feature/refactor code

ivangonzalez224 committed:
Update the paths to the deployment api app

ivangonzalez224 committed:
Include the demo live section in the readme file

ivangonzalez224 committed:
Merge pull request #38 from ivangonzalez224/feature/deployment

Deploy the app

ivangonzalez224 committed:
Update paths to connect with local server

ivangonzalez224 committed:
Merge pull request #39 from ivangonzalez224/feature/final-updates

Update paths
